x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/wrapper.tsx

```diff
@@ -50,9 +56,17 @@ interface AiForSOCAlertsTableProps {
  * It renders a loading skeleton while packages are being fetched and while the dataView is being created.
  */
 export const AiForSOCAlertsTable = memo(({ id, onLoaded, query }: AiForSOCAlertsTableProps) => {
-  const { data } = useKibana().services;
-  const [dataView, setDataView] = useState<DataView | undefined>(undefined);
-  const [dataViewLoading, setDataViewLoading] = useState<boolean>(true);
+  const { dataView: oldDataView, loading: oldDataViewLoading } = useCreateDataView({
+    dataViewSpec: AI4SOC_DATA_VIEW_SPEC,
+  });
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+  // TODO: use alert only data view when it is ready
+  // https://github.com/elastic/security-team/issues/12589
+  const { dataView: experimentalDataView, status } = useDataView(SourcererScopeName.detections);
+
+  const loading = newDataViewPickerEnabled ? status !== 'ready' : oldDataViewLoading;
+  const dataView = newDataViewPickerEnabled ? experimentalDataView : oldDataView;
```

Comment:
This code is repeated in a few places, what do you think about extracting these lines into a hook. We could put it under this path `x-pack/solutions/security/plugins/security_solution/public/detections/hooks/alert_summary` and it could return
 ```typescript
dataView: DataView;
loading: boolean;
```
Wdyt?

Reply:
I would simplify a bit the unit tests of the components themselves, as you'd only have to mock the hook. Less conditions to test in the components, just test those conditions in the newly created hook tests

Reply:
Fair, I'm ok with keeping what we have now! Looking forward to seeing all of this removed soon :)